engine/src/known_types.rs

```diff
@@ -475,6 +475,15 @@ fn create_type_database() -> TypeDatabase {
         false,
     ));
 
+    db.insert(TypeDetails::new(
+        "core::pin::Pin",
+        "Pin",
+        Behavior::RustByValue, // because this is actually Pin<&something>
+        None,
+        true,
+        false,
```

Comment:
Thanks, I'm doing a proper review of this now the tests all pass.

My only concern is as follows.

Here you're adding a whole extra entry to the `known_types.rs` database.

That makes me a bit uncomfortable because it's actually the _same_ type as `std::pin::Pin`. I don't know what might go wrong by having an _extra_ entry here, and all the tests pass, so perhaps it's OK - but it still feels semantically wrong.

Did you try instead:
* have a single entry, where the main Rust name is `core::pin::Pin`
* but instead of `None`, specify `Some("std::pin::Pin")`

If that works and all the tests pass, I'd prefer it? It just feels more semantically correct.

If it doesn't work, I'd like to know why not, because it might help me become comfortable with having two entries here instead of one!

Apart from that, this all LGTM.

Reply:
> it still feels semantically wrong

That's fair. I wasn't familiar with the semantics here. I hadn't actually checked what field that "none" was for. (small quibble: i feel like a Vec may be more appropriate? but I guess it's fine for now.)

Technically we probably don't need the entry for `std::pin::Pin` anymore, as we should no longer be generating code that uses that path. I think the tests pass fine without it. I'll move it to a `extra_non_canonical_name`, but I'm not sure how to even test this case. (maybe we want to remove it entirely so that references to std don't sneak back in?)